### ISSUE TITLE
Flags cannot succeed a StringArrayArgument

### ISSUE THREAD

**Issue by AlessioGr** [Incendo/cloud#321]
![image](https://user-images.githubusercontent.com/70709113/143658307-28035ecc-d4ab-4cde-b51c-9a26e8db8eae.png)

I also have a flag called `--nametag_containsany`. However, it is not being auto-completed (I also doesn't work if I type it manually) when I make `--nametag_equals` a StringArrayArgument:

```java
        nametag_equals = CommandFlag
                .newBuilder("nametag_equals")
                .withArgument(StringArrayArgument.of("nametag_equals",
                                (context, lastString) -> {
                                    ArrayList<String> completions = new ArrayList<>();
                                    completions.add("<nametag_equals flag value>");
                                    return completions;
                                }
                        ))
                .build();
```

The expected behaviour would be to have both flags at the same time work like `/[...] --nametag_equals one two three --nametag_containsany four five six`

**Comment by Citymonstret** [comment]
StringArrayArgument (similar to Greedy strings) never stops consuming input. Thus, we cannot really know whether "--secondflag" should be consumed by the StringArrayArgument/greedy, or the flag argument.

The solution would be to implement something like https://github.com/Incendo/cloud/issues/218.

**Comment by Citymonstret** [comment]
This has been "fixed" in 1.7.0 with the introduction of flag yielding arguments. See https://github.com/Incendo/cloud/blob/1.7.0-dev/cloud-core/src/test/java/cloud/commandframework/issue/Issue321.java.
